hearsttv/video-auto-clipping
# Htvds bug fix

# Jira URL
https://thetower.atlassian.net/browse/HTVDS-168

# Description
This PR includes:
* Minor refactorings
* New exception `autoclip.uplynk.exceptions.InvalidExternalD`:
  when a newscast retrieved from Uplynk has an empty external ID.
* Clipping lambda throws the exception `autoclip.uplynk.exceptions.InvalidExternalD`
  The goal is to send to the SQS retry queue events that trigger this exception, 
  so that in a later attempt the external ID may be correctly set in Uplynk.

# Testing Instructions
## Unit test
Run tox

## Feature Test
1. Open [this newscast](https://cms.uplynk.com/static/cms2/index.html#/content/a188e031443f4b178b7c125aa3228d31/) details page.
2. Delete the value of the field `External ID` and click `Save`;
3. Execute the local or dev test procedures described below;
4. Check in the logs that the `InvalidExternalID` exception has been thrown;
5. In Uplynk, return the value of the field `External ID` to `200702_120007_1_khtv` and click `Save`.
6. If testing locally, then execute the test procedure once again. If testing in dev, wait for the DLQ retry delay.
7. Check in the logs that the clip draft creation has been successful.

### Local Test
* Open the Jupyter Notebook 
* Ensure the following variables are set as follows in the first cell of the notebook:
  ```python
  STATION = 'khtv'
  NEWS_DATE = '2020-07-02'
  XML_FILE_NAME = '120006'
  ```
* Run the notebook (you will need to set the environmental variable `VAULT_PASSWORD` first)

### Dev Test
Drop into S3 the XML file `tests/data/khtv/2020-07-02/120006.xml` with a file name 
not yet existing in the destination directory. E.g., increase/decrease the number of the 
XML name by 1. This is necessary to trigger the event message.

Here is an example the AWS CLI command (from the repo root):
`aws s3 cp tests/data/khtv/2020-07-02/120006.xml s3://enps-bucket/khtv/2020-07-02/120007.xml`
